### DISCUSSION TITLE
What is proper way to re-render an item from an observable list?

### DISCUSSION THREAD

**Discussion by Albert-Gao** [Nozbe/WatermelonDB#1428] [Q&A]
<img width="647" alt="image" src="https://user-images.githubusercontent.com/18282328/199608828-ba185e1d-90af-4e30-b321-abb0815d84ef.png">

The above code works well for adding and removing, however, if I update the item within this list, it will not get updated until I refresh the screen.

I think one way is to subscribe to this item on the item component level. 
Will it causes performance problem for a large list, it is sort of a double subscription in which the list subscribed once, and subscribed again at the item level.

**Comment by batuhansahan** [comment]
observeWithColumns is your solution check it out in docs

**Comment by ororsatti** [comment]
Are you sure it doesnt get triggered? From what I have seen wm triggers the observed queries every time something change in the desired table.

I think its more of a thing in how react compares lists changes as props. 
On any way, you can just observe each item individually, from what I have seen it doesnt cause a performance issue, you should always test it yourself tho.

**Comment by radex** [comment]
`ItemBase` should be wrapped with an `withObservables` that subscribes to `item`'s changes.
